samples/modules/thrift/hello_server/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+# Copyright 2022 Meta
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(thrift_hello_server)
+
+FILE(GLOB app_sources
+    src/*.cpp
+)
+
+include(${ZEPHYR_BASE}/cmake/util/thrift.cmake)
+
+set(generated_sources "")
+set(gen_dir ${ZEPHYR_BINARY_DIR}/misc/generated/thrift_hello)
+list(APPEND generated_sources ${gen_dir}/gen-cpp/hello_types.h)
+list(APPEND generated_sources ${gen_dir}/gen-cpp/Hello.cpp)
+list(APPEND generated_sources ${gen_dir}/gen-cpp/Hello.h)
+list(APPEND app_sources ${generated_sources})
+
+thrift(
+    app
+    cpp
+    :no_skeleton
+    ${gen_dir}
+    "${ZEPHYR_BASE}/modules/thrift/misc/hello.thrift"
```

Comment:
This should be part of the samples, can have e.g. `samples/modules/thrift/hello_world` folder with the 4 applications then in this folder and have this shared file in the hello_world folder so that everything is contained and easy for people to see and understand.